src/api/app/mixins/has_attributes.rb

```diff
@@ -40,20 +40,14 @@ def store_attribute(namespace, name, values, issues, binary = nil)
 
     # update or create attribute entry
     changed = false
-    begin
-      a = find_attribute(namespace, name, binary)
-    rescue AttributeFindError => e
-      raise AttributeSaveError, e
-    end
+    a = find_attribute(namespace, name, binary)
     if a.nil?
       # create the new attribute
       a = Attrib.new(attrib_type: attrib_type, binary: binary)
       a.project = self if is_a? Project
       a.package = self if is_a? Package
-      if a.attrib_type.value_count
-        a.attrib_type.value_count.times do |i|
-          a.values.build(position: i, value: values[i])
-        end
+      (a.attrib_type.value_count || 0).times do |i|
+        a.values.build(position: i, value: values[i])
```

Comment:
I am fine with the change, but this would change our API, right? I would at least document this in the release notes.

Reply:
how so? by allowing value resorts you mean?

Reply:
I was referring to the whole [commit](995f93922874329940bca28dfcc7b6d9338a37ce), not that particular line. Just had to attach my comment somewhere.

But it looks like the API [changes](https://github.com/openSUSE/open-build-service/pull/5165/commits/d33b7c53015b4a1b4aad81fc2c3275af8fcbb206#diff-43a2207982cf719e0794f875b2b7c4e4R477). As said I personally think this is okay, since ithe change makes sense. But I would document this.